.github/workflows/build.yml

```diff
@@ -134,12 +135,16 @@ jobs:
           distribution: temurin
           java-version: 17
 
-      - uses: gradle/actions/setup-gradle@v3
+      - name: Set up gradle
+        uses: gradle/actions/setup-gradle@v4
+        # skipping release branches because the versions in those branches are not snapshots
+        # (also this skips pull requests)
+        if: ${{ github.ref_name == 'main' && github.repository == 'open-telemetry/opentelemetry-java' }}
```

Comment:
This duplication is unfortunate.. Nice to only have to write the condition once

Reply:
Yeah it's true. Maybe down the road we can use a variable for this? I dunno.